### ISSUE TITLE
sensor ringer_mode is not updated.

### ISSUE THREAD

**Issue by realjax** [home-assistant/android#3257]
**Home Assistant Android version:**
2022,12.0-full

**Android version:**
12.skq1.211006.001
 
**Phone model:**
Xiaomi Poco F3

**Home Assistant version:**
2023.1.7

**Last working Home Assistant release (if known):**

**Description of problem:**
The sensor ringer_mode is not updated. It eihjer stays on 'silent' when it shouldn't be, or it says unknown. Changing the status on the phone has no effect at all.

**Comment by dshokouhi** [comment]
Please do not remove any part of the issue template, all requested information is important to help you out. Without the logs that you skipped over we cannot proceed with troubleshooting. You will need to reproduce the issue and get us the on device logs from Settings > Companion App > show and share logs.

**Comment by realjax** [comment]
As fas as I can see that log request is not in the template? Nevertheless, here's the part that should have any info. (I have removed location info) At 18:53 the ringer mode was changed to silent and then back again...
`
52:51.430 13145 13326 I OpenGLRenderer: Davey! duration=9222767939575ms; Flags=0, FrameTimelineVsyncId=13560651, IntendedVsync=604097278758737, Vsync=604097278758737, InputEventId=0, HandleInputStart=604097281514496, AnimationStart=604097281519548, PerformTraversalsStart=604097281522673, DrawStart=604097281896058, FrameDeadline=604097312092069, FrameInterval=604097281452881, FrameStartTime=16666666, SyncQueued=604097282879079, SyncStart=604097282916943, IssueDrawCommandsStart=604097283254183, SwapBuffers=604097291171579, FrameCompleted=9223372036854775807, DequeueBufferDuration=81979, QueueBufferDuration=1898958, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604097294648246, DisplayPresentTime=0, 
01-23 18:52:53.926 13145 13326 I OpenGLRenderer: Davey! duration=9222767937079ms; Flags=0, FrameTimelineVsyncId=13560885, IntendedVsync=604099774712587, Vsync=604099774712587, InputEventId=0, HandleInputStart=604099777679130, AnimationStart=604099777684182, PerformTraversalsStart=604099777687307, DrawStart=604099778053453, FrameDeadline=604099808045919, FrameInterval=604099777617620, FrameStartTime=16666666, SyncQueued=604099779121265, SyncStart=604099779503817, IssueDrawCommandsStart=604099779794495, SwapBuffers=604099788317515, FrameCompleted=9223372036854775807, DequeueBufferDuration=97969, QueueBufferDuration=1721459, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604099791245015, DisplayPresentTime=0, 
01-23 18:53:08.831 13145 13326 I OpenGLRenderer: Davey! duration=9222767922172ms; Flags=0, FrameTimelineVsyncId=13561275, IntendedVsync=604114681943638, Vsync=604114681943638, InputEventId=0, HandleInputStart=604114685539593, AnimationStart=604114685543551, PerformTraversalsStart=604114685545531, DrawStart=604114685847093, FrameDeadline=604114698610304, FrameInterval=604114685493760, FrameStartTime=16666666, SyncQueued=604114686648499, SyncStart=604114686717353, IssueDrawCommandsStart=604114687280635, SwapBuffers=604114693164385, FrameCompleted=9223372036854775807, DequeueBufferDuration=61354, QueueBufferDuration=1238854, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604114695242197, DisplayPresentTime=0, 
01-23 18:53:15.052 13145 13145 D MIUIInput: [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=604120922, downTime=604120922, deviceId=-1, source=0x101, displayId=0 }
01-23 18:53:15.055 13145 13145 D MIUIInput: [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=604120922, downTime=604120922, deviceId=-1, source=0x101, displayId=0 }
01-23 18:53:15.067 13145 13145 D UrlRepository: Using external URL
01-23 18:53:16.270 13145 13145 D LocBroadcastReceiver: Received location update.
01-23 18:53:16.271 13145 13145 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-23 18:53:16.271 13145 13145 D LocBroadcastReceiver: Last Location: 
01-23 18:53:16.271 13145 13145 D LocBroadcastReceiver: Coords:(x,y)
01-23 18:53:16.271 13145 13145 D LocBroadcastReceiver: Accuracy: 12.947
01-23 18:53:16.271 13145 13145 D LocBroadcastReceiver: Bearing: 0.0
01-23 18:53:16.274 13145 13145 D UrlRepository: localUrl is: true, usesInternalSsid is: true, usesWifi is: true
01-23 18:53:16.275 13145 13145 D LocBroadcastReceiver: Begin evaluating if location update should be skipped
01-23 18:53:16.275 13145 13145 D LocBroadcastReceiver: Received location that is 86 milliseconds old, 1674496396189 compared to 1674496396275 with source fused
01-23 18:53:16.284 13145 13304 D UrlRepository: localUrl is: true, usesInternalSsid is: true, usesWifi is: true
01-23 18:53:16.360 13145 13304 D LocBroadcastReceiver: Location update sent successfully as exact
01-23 18:53:20.978 13145 13145 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1674496339640, currentMillis: 1674496400978)
01-23 18:53:22.236 13145 13304 D LogcatReader: Read logcat for pid 13145
01-23 18:53:22.477 13145 13304 D LogcatReader: Done reading logcat for pid 13145
01-23 18:53:27.260 13145 13339 I OpenGLRenderer: Davey! duration=9222767903742ms; Flags=0, FrameTimelineVsyncId=13562662, IntendedVsync=604133112697195, Vsync=604133112697195, InputEventId=576891792, HandleInputStart=604133112954326, AnimationStart=604133113570732, PerformTraversalsStart=604133113661878, DrawStart=604133113859742, FrameDeadline=604133129363861, FrameInterval=604133112931044, FrameStartTime=16666666, SyncQueued=604133114316982, SyncStart=604133114378128, IssueDrawCommandsStart=604133114698701, SwapBuffers=604133122095888, FrameCompleted=9223372036854775807, DequeueBufferDuration=65365, QueueBufferDuration=2191511, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604133125131513, DisplayPresentTime=0, 
01-23 18:53:28.466 13145 14433 I OpenGLRenderer: Davey! duration=9222767902536ms; Flags=0, FrameTimelineVsyncId=13562785, IntendedVsync=604134318231015, Vsync=604134318231015, InputEventId=0, HandleInputStart=604134319216252, AnimationStart=604134319219065, PerformTraversalsStart=604134319220888, DrawStart=604134319397763, FrameDeadline=604134351564347, FrameInterval=604134319184950, FrameStartTime=16666666, SyncQueued=604134319985888, SyncStart=604134320067919, IssueDrawCommandsStart=604134320360627, SwapBuffers=604134329560836, FrameCompleted=9223372036854775807, DequeueBufferDuration=106823, QueueBufferDuration=1330833, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604134331876981, DisplayPresentTime=0, 
01-23 18:53:38.332 13145 13339 I OpenGLRenderer: Davey! duration=9222767892670ms; Flags=0, FrameTimelineVsyncId=13563283, IntendedVsync=604144184568707, Vsync=604144184568707, InputEventId=0, HandleInputStart=604144186206248, AnimationStart=604144186211144, PerformTraversalsStart=604144186214165, DrawStart=604144186615676, FrameDeadline=604144217902039, FrameInterval=604144186143905, FrameStartTime=16666666, SyncQueued=604144187590103, SyncStart=604144187672811, IssueDrawCommandsStart=604144187946873, SwapBuffers=604144192749321, FrameCompleted=9223372036854775807, DequeueBufferDuration=51406, QueueBufferDuration=1090729, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604144194423436, DisplayPresentTime=0, 
01-23 18:53:42.843 13145 14437 I OpenGLRenderer: Davey! duration=9222767888157ms; Flags=0, FrameTimelineVsyncId=13563432, IntendedVsync=604148696753948, Vsync=604148696753948, InputEventId=0, HandleInputStart=604148697528382, AnimationStart=604148697531403, PerformTraversalsStart=604148697605622, DrawStart=604148697891663, FrameDeadline=604148730087280, FrameInterval=604148697473851, FrameStartTime=16666666, SyncQueued=604148699272601, SyncStart=604148699394736, IssueDrawCommandsStart=604148701025518, SwapBuffers=604148710710570, FrameCompleted=9223372036854775807, DequeueBufferDuration=57187, QueueBufferDuration=2121718, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604148714073070, DisplayPresentTime=0, 
01-23 18:53:55.290 13145 14437 I OpenGLRenderer: Davey! duration=9222767875712ms; Flags=0, FrameTimelineVsyncId=13563970, IntendedVsync=604161142018119, Vsync=604161142018119, InputEventId=0, HandleInputStart=604161144486346, AnimationStart=604161144491242, PerformTraversalsStart=604161144493846, DrawStart=604161144909784, FrameDeadline=604161158684785, FrameInterval=604161144419784, FrameStartTime=16666666, SyncQueued=604161146312023, SyncStart=604161146401190, IssueDrawCommandsStart=604161146693065, SwapBuffers=604161151124367, FrameCompleted=9223372036854775807, DequeueBufferDuration=56198, QueueBufferDuration=1393750, GpuCompleted=9223372036854775807, SwapBuffersCompleted=604161153182648, DisplayPresentTime=0, 
01-23 18:54:07.584 13145 13145 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1674496339640, currentMillis: 1674496447584)
01-23 18:54:10.511 13145 13145 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1674496339640, currentMillis: 1674496450511)
01-23 18:54:13.946 13145 13145 D IntegrationRepository: isAppLocked(): false. (LockEnabled: false, appActive: false, expireMillis: 1674496339640, currentMillis: 1674496453946)
01-23 18:54:15.662 13145 13145 D MIUIInput: [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=604181531, downTime=604181531, deviceId=-1, source=0x101, displayId=0 }
01-23 18:54:15.664 13145 13145 D MIUIInput: [KeyEvent] ViewRootImpl KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x8, repeatCount=0, eventTime=604181531, downTime=604181531, deviceId=-1, source=0x101, displayId=0 }
01-23 18:54:15.676 13145 13145 D UrlRepository: Using external URL
01-23 18:54:16.304 13145 13145 D LocBroadcastReceiver: Received location update.
01-23 18:54:16.305 13145 13145 D ForegrndServiceLauncher: Check if service HighAccuracyLocationService is running. Service running = false
01-23 18:54:16.305 13145 13145 D LocBroadcastReceiver: Last Location: 
01-23 18:54:16.305 13145 13145 D LocBroadcastReceiver: Coords:(x,y)
`

**Comment by dshokouhi** [comment]
These logs are only 1 minute long, did you reproduce the issue by changing the ringer mode on the device in this short time frame? These logs are missing the intent that we listen for when ringer mode changes to update the state.

Are you certain you are adjusting the ringer mode and not DND? Some devices may mix those up. And also does your device have a hardware switch that also controls ringer mode? Really we depend on the device to tell us what ringer mode is in and report that, if the device does not report properly in the API then we will also not report properly.

https://github.com/home-assistant/android/blob/master/common/src/main/java/io/homeassistant/companion/android/common/sensors/AudioSensorManager.kt#L180-L185

Try this, force stop the app and open it then go to Settings > Companion App > Manage Sensors > Ringer Mode sensor and from there change the ringer mode on the device to Normal > Vibrate > Silent then check the logs. Please also share a screenshot when you change the ringer mode and the sensor screen shows the incorrect state.

**Comment by realjax** [comment]
Thanks for your help! Yes the log is 1 minute long because I don't feel.like having to remove 100 location updates manually. And yes I did change back and forth in that 1 minute. And No I did not toggle DND. And yes, I was missing the intent too. 😊 Like I said nothing happens. I have force quit the app and retried. Even retried from the same position like you said.. will try again though. It might be like you said and the phone isn't reporting correctly. Wouldn't be surprised with xiaomis MIUI.
